code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -678,93 +678,38 @@ var/global/list/damage_icon_parts = list()
 
 	overlays_standing[HO_TAIL_OVER_LAYER] =  null
 	overlays_standing[HO_TAIL_UNDER_LAYER] = null
-	var/tail_state = bodytype.get_tail(src)
-	if(tail_state && (!wear_suit || !(wear_suit.flags_inv & HIDETAIL)))
-		var/icon/tail_s = get_tail_icon()
-		overlays_standing[(dir == NORTH) ? HO_TAIL_OVER_LAYER : HO_TAIL_UNDER_LAYER] = image(tail_s, icon_state = "[tail_state]_s")
-		animate_tail_reset(0)
+	var/obj/item/organ/external/tail/tail_organ = bodytype.get_tail(src)
+	if(tail_organ && !tail_organ.is_stump() && (!wear_suit || !(wear_suit.flags_inv & HIDETAIL)))
+		var/icon/tail_icon = get_tail_icon(tail_organ)
+		overlays_standing[(dir == NORTH) ? HO_TAIL_OVER_LAYER : HO_TAIL_UNDER_LAYER] = image(tail_icon)
 
 	if(update_icons)
 		update_icon()
 
-/mob/living/carbon/human/proc/get_tail_icon()
-	var/icon_key = "[bodytype.get_icon_cache_uid(src)][skin_colour][hair_colour]"
+/mob/living/carbon/human/proc/get_tail_icon(var/obj/item/organ/external/tail/tail_organ)
+	var/icon_key = "[bodytype.get_icon_cache_uid(src)][skin_colour][hair_colour][bodytype.name]"
```

Comment:
What's the purpose of adding `bodytype.name`? `get_icon_cache_uid()` should already guarantee the bodytype is unique.